python/python/lance/dataset.py

```diff
@@ -112,16 +112,20 @@ def execute(self, data_obj: ReaderLike, *, schema: Optional[pa.Schema] = None):
 
     # These next three overrides exist only to document the methods
 
-    def when_matched_update_all(self):
+    def when_matched_update_all(self, condition: Optional[str] = None):
         """
         Configure the operation to update matched rows
 
         After this method is called, when the merge insert operation executes,
         any rows that match both the source table and the target table will be
         updated.  The rows from the target table will be removed and the rows
         from the source table will be added.
+
+        An optional condition may be specified.  This should be an SQL filter
```

Comment:
Also, should we mention to use the `target.` and `source.` prefixes in column references?

Reply:
Definitely.  I will add that.

Reply:
Added